engine/runtime/src/main/java/org/enso/interpreter/node/callable/ExecuteCallNode.java

```diff
@@ -46,20 +50,45 @@ public static ExecuteCallNode build() {
    * @param callNode the cached call node for {@code cachedTarget}
    * @return the result of executing {@code function} on {@code arguments}
    */
-  @Specialization(guards = "function.getCallTarget() == cachedTarget")
+  @Specialization(guards = {
+    "function.getCallTarget() == cachedTarget",
+    "callNode != null"
+  })
   protected Object callDirect(
+      VirtualFrame frame,
       Function function,
       CallerInfo callerInfo,
       Object state,
       Object[] arguments,
       @Cached("function.getCallTarget()") RootCallTarget cachedTarget,
-      @Cached("createCallNode(cachedTarget)") DirectCallNode callNode) {
-    return callNode.call(
-        Function.ArgumentsHelper.buildArguments(function, callerInfo, state, arguments));
+      @Cached("createInlineableNode(cachedTarget)") InlineableNode callNode) {
+    var args = Function.ArgumentsHelper.buildArguments(function, callerInfo, state, arguments);
+    return callNode.call(frame, args);
+  }
+
+  @Specialization(guards = {
+    "function.getCallTarget() == cachedTarget",
+  })
+  protected Object callDirect(
```

Comment:
`InlineableNode` no longer subclasses `DirectCallNode` so whoever wants to use it, needs two `@Specialization`s. One for `InlineableNode` and one for `DirectCallNode` .